### ISSUE TITLE
BigtableUploadService fails to upload when `--no-snapshot-fetch` is not set

### ISSUE THREAD

**Issue by buffalu** [solana-labs/solana#26080]
#### Problem
When one starts an RPC server for the first time without the `--no-snapshot-fetch` command line provided, the validator will fail to upload bigtable slots as its stuck on slot 0 and continues looping.

```
[2022-06-20T15:36:01.050475190Z INFO  solana_ledger::bigtable_upload] Loading ledger slots starting at 0...
[2022-06-20T15:36:01.050551604Z WARN  solana_ledger::bigtable_upload_service] bigtable: upload_confirmed_blocks: Ledger has no slots from 0 to 192
```

#### Proposed Solution
The validator should load from the first actual slot in blockstore and not get stuck on slot 0.

**Comment by CriesofCarrots** [comment]
What version do you see this on @buffalu ? This is a known issue, but expected to be fixed by this: https://github.com/solana-labs/solana/pull/25575

**Comment by buffalu** [comment]
> What version do you see this on @buffalu ? This is a known issue, but expected to be fixed by this: #25575

just saw it on 1.10.26. 

we got forked off w/ the update, so tried to update and restart with current ledger but that didn't work.
wiped ledger and restarted without the `--no-snapshot-fetch` flag and noticed it kept getting stuck on that line.
let it run for a few minutes and it didn't fix itself, so ended up restarting the server with that flag re-enabled and its okay now

**Comment by CriesofCarrots** [comment]
@buffalu , sorry for the delay, but I think I have this one figured. Seems like the bigtable upload service starts trying to read blockstore before the snapshot root has been written. I have a hacky fix here: https://github.com/CriesofCarrots/solana/commit/96a16daaee3663a8805befdb27d6fa842e073f96
Would you be able to test it, by chance?
